onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -865,9 +866,19 @@ Status QnnBackendManager::SetupBackend(const logging::Logger& logger,
     LOGS(logger, VERBOSE) << "InitializeProfiling succeed.";
   }
 
+  bool enable_htp_weight_sharing = false;
+  // weight sharing only available with offline generation on x64 platform, not available on the real device
+#if defined(__aarch64__) || defined(_M_ARM64)
+  ORT_UNUSED_PARAMETER(share_ep_contexts);
```

Comment:
Should we log a warning (or maybe return an error) if `share_ep_contexts && !load_from_cached_context` is true on the real device?